apps/files_sharing/lib/Controller/ShareAPIController.php

```diff
@@ -718,7 +727,7 @@ public function createShare(
 			}
 
 			// Only share by mail have a recipient
-			if (is_string($shareWith) && $shareType === IShare::TYPE_EMAIL) {
+			if (!empty($shareWith) && !is_null($shareWith) && $shareType === IShare::TYPE_EMAIL) {
```

Comment:
```suggestion
			if ($shareType === IShare::TYPE_EMAIL) {
```
You don't need these checks because the comparison already handles these cases (and psalm doesn't like it either)